tests/test_grainstats_minicircle.py

```diff
@@ -12,28 +12,34 @@
 RESOURCES = BASE_DIR / "tests" / "resources"
 
 
-def test_grainstats(minicircle_grainstats: GrainStats, minicircle_grainstats_20220526: pd.DataFrame) -> None:
-    """Test the overall GrainStats class."""
+def test_grainstats_regression(regtest, minicircle_grainstats: GrainStats) -> None:
+    """Regression tests for grainstats."""
     statistics = minicircle_grainstats.calculate_stats()
+    print(statistics["statistics"].to_string(), file=regtest)
```

Comment:
Is the print statement part of troubleshooting or is this creating the output file?

Reply:
This is one of three ways of `pytest-regtest` to save its target output. The other ways are to use...

```
        # alternative method to record output:
        regtest.write("done")

        # or using a context manager:
        with regtest:
            print("this will be recorded")
```

It fails the first time you run the test showing the output and that its compared to nothing. You then have to run it with the `--regtest-reset` option and this results in the `print()` statement writing to file the expected output.

I opted for the `print(statistics["statistics"].to_string(), file=regtest)` because I wanted to forcibly print all of the data frame (using the `.to_string()` method) so the whole data frame could be compared.

Home page for [pytest-regtest](https://gitlab.com/uweschmitt/pytest-regtest) for reference.